### PR TITLE
net-p2p/qbittorrent: add 4.6.0

### PR DESCRIPTION
**Pull request by listout**

### CODE REVIEW

**Comment by xbt573** [comment]
lgtm, thanks
i have some problems with invisible tray icon on 4.6.0 qt5 build, but it seems like i'm the only one with this problem, because other people i asked to test didn't report this issue

File: net-p2p/qbittorrent/qbittorrent-4.6.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+			dev-qt/qtsvg:6
+		)
+	)
+	qt6? ( dev-qt/qtbase:6[network,ssl,sql,sqlite,xml(+)] )
+	!qt6? (
+		dev-qt/qtcore:5
+		dev-qt/qtnetwork:5[ssl]
+		dev-qt/qtsql:5[sqlite]
+		dev-qt/qtxml:5
+	)
```
Comment (juippis) [inline net-p2p/qbittorrent/qbittorrent-4.6.0.ebuild:50]: These should probably be grouped under `gui` conditions, as `-gui +qt6` does nothing right?

File: net-p2p/qbittorrent/qbittorrent-4.6.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+RESTRICT="!test? ( test )"
+REQUIRED_USE="|| ( gui webui )"
+
+RDEPEND="
+	>=dev-libs/openssl-1.1.1:=
+	>=net-libs/libtorrent-rasterbar-1.2.18:=
+	>=sys-libs/zlib-1.2.11
+	virtual/libiconv
+	gui? (
+		dev-libs/geoip
```
Comment (juippis) [inline net-p2p/qbittorrent/qbittorrent-4.6.0.ebuild:32]: Is this really mandatory...?

File: net-p2p/qbittorrent/qbittorrent-4.6.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+	)
+	qt6? ( dev-qt/qtbase:6[network,ssl,sql,sqlite,xml(+)] )
+	!qt6? (
+		dev-qt/qtcore:5
+		dev-qt/qtnetwork:5[ssl]
+		dev-qt/qtsql:5[sqlite]
+		dev-qt/qtxml:5
+	)
+	acct-group/qbittorrent
+	acct-user/qbittorrent"
```
Comment (juippis) [inline net-p2p/qbittorrent/qbittorrent-4.6.0.ebuild:52]: Should be moved before the conditional deps.

File: net-p2p/qbittorrent/qbittorrent-4.6.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+			# More verbose build logs are preferable for bug reports
+			-DVERBOSE_CONFIGURE=ON
+
+			-DQT6=$(usex qt6)
+
+			-DWEBUI=$(usex webui)
+
+			-DTESTING=$(usex test)
+		)
```
Comment (juippis) [inline net-p2p/qbittorrent/qbittorrent-4.6.0.ebuild:89]: Weird whitespacing compared to other `mycmakeargs` blocks.

File: net-p2p/qbittorrent/qbittorrent-4.6.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+	inherit git-r3
+else
+	SRC_URI="https://github.com/qbittorrent/qBittorrent/archive/release-${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~riscv ~x86"
+	S="${WORKDIR}"/qBittorrent-release-${PV}
+fi
+
+LICENSE="GPL-2"
+SLOT="0"
+IUSE="+dbus +gui qt6 test webui systemd"
```
Comment (juippis) [inline net-p2p/qbittorrent/qbittorrent-4.6.0.ebuild:22]:
```suggestion
IUSE="+dbus +gui qt6 systemd test webui"
```

File: net-p2p/qbittorrent/qbittorrent-4.6.0.ebuild (hunk 2 above)
Comment (listout) [reply net-p2p/qbittorrent/qbittorrent-4.6.0.ebuild:34]: Perhaps @xbt573 could comment/suggest something.

Comment (xbt573) [reply net-p2p/qbittorrent/qbittorrent-4.6.0.ebuild:34]: zlib is used in app
geoip database downloads during runtime if i don't miss something, at least i can't find it in CMakeLists
can't find libiconv too, so i guess it's useless in ebuild

Comment (juippis) [reply net-p2p/qbittorrent/qbittorrent-4.6.0.ebuild:34]: It only seems to link to the library while compiling with `+webui`. Now I don't doubt it uses it with `+gui` too, but something you could check and verify.

File: net-p2p/qbittorrent/qbittorrent-4.6.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+RESTRICT="!test? ( test )"
+REQUIRED_USE="|| ( gui webui )"
+
+RDEPEND="
+	>=dev-libs/openssl-1.1.1:=
+	>=net-libs/libtorrent-rasterbar-1.2.18:=
+	>=sys-libs/zlib-1.2.11
+	virtual/libiconv
+	acct-group/qbittorrent
+	acct-user/qbittorrent
```
Comment (juippis) [inline net-p2p/qbittorrent/qbittorrent-4.6.0.ebuild:32]:
```suggestion
	acct-group/qbittorrent
	acct-user/qbittorrent
	>=dev-libs/openssl-1.1.1:=
	>=net-libs/libtorrent-rasterbar-1.2.18:=
	>=sys-libs/zlib-1.2.11
	virtual/libiconv
```

**Review by juippis — APPROVED** [review]
Even with `+webui -gui` it wanted some qt5network package,
```
CMake Error at /usr/lib64/cmake/Qt5/Qt5Config.cmake:28 (find_package):
  Could not find a package configuration file provided by "Qt5Network" with
  any of the following names:

    Qt5NetworkConfig.cmake
    qt5network-config.cmake

  Add the installation prefix of "Qt5Network" to CMAKE_PREFIX_PATH or set
  "Qt5Network_DIR" to a directory containing one of the above files.  If
  "Qt5Network" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  cmake/Modules/CheckPackages.cmake:59 (find_package)
  CMakeLists.txt:58 (include)


-- Configuring incomplete, errors occurred!
 * ERROR: net-p2p/qbittorrent-4.6.0::gentoo failed (configure phase):
 *   cmake failed
```
So looks like at least that is required.

**Pull request closed without merging**